### DISCUSSION TITLE
Doesn't display anything when `composer.json` is missing

### DISCUSSION THREAD

**Discussion by gander** [composer-unused/composer-unused#247] [Ideas]
When I run the application in a directory that does not contain composer.json, the application displays nothing. Never. It only returns the error code. Only when I put the composer.json file in a given directory, the application starts displaying the text.

![image](https://user-images.githubusercontent.com/421597/146048313-0cb40976-d36d-4b27-9616-fe72ace64f19.png)

**Comment by icanhazstring** [comment]
Good point. Seems like a little nice addition to output something like: Not a composer project.

**Reply by gander** [reply]
For both cases, when using the -h option, I should see the same help.  Only by running the main command in a directory without composer.json, I should see an error message.

**Reply by gander** [reply]
Now I can't see anything when I run in such a directory.  And I don't know if the program is not working or if I am doing something wrong.  Not everyone will come up with the idea to check the exit code.

**Reply by icanhazstring** [reply]
The reason is this: https://github.com/composer-unused/composer-unused/blob/main/src/Console/Command/UnusedCommand.php#L90

As the command is a composer plugin, you can't run it right now without an instance of composer. Without a composer.json no instance.

**Reply by gander** [reply]
This error code (1) comes from inside the original composer code, not composer-unused:
https://github.com/composer-unused/composer-unused/blob/main/bin/composer-unused#L44
https://github.com/composer/composer/blob/2.1.14/src/Composer/Console/Application.php#L438

The missing error message is the result of this (`NullIO`):
https://github.com/composer/composer/blob/2.1.14/src/Composer/Console/Application.php#L112

---
Should be displayed by composer:
```
Composer could not find a composer.json file in /www/app
To initialize a project, please create a composer.json file. See https://getcomposer.org/basic-usage
```

**Reply by icanhazstring** [reply]
I know, that is what I am saying. At the moment the binary is faking a console application for composer.

So if there is no composer.json nothing can work. I will remove the composer dependency itself at some point.

**Reply by gander** [reply]
Temporarily you can give such a solution. If all goes well, `$this->composer` is generated internally anyway, and the next call to this method returns it. If it goes wrong, you catch the message from composer and display it.

***bin/composer-unused***
```php
    try {
        $application->setCatchExceptions(false);
        $application->getComposer();
        $application->setCatchExceptions(true);
    } catch (\Exception $exception) {
        fwrite(STDERR, $exception->getMessage() . PHP_EOL);
        exit(1);
    }
```

Result:
```
/tmp$: /www/bin/composer-unused; echo $?

Composer could not find a composer.json file in /tmp
To initialize a project, please create a composer.json file. See https://getcomposer.org/basic-usage
1
```

**Reply by gander** [reply]
Or create your own message when `composer.json` is missing as with undefined `UNUSED_COMPOSER_INSTALL` constant.

**Reply by icanhazstring** [reply]
Hi @gander this has been resolved with 0.8 of composer-unused. :)
